.travis.yml

```diff
@@ -17,21 +24,160 @@ gemfile:
+  - gemfiles/6.0.gemfile
+  - gemfiles/6.1.gemfile
+  - gemfiles/7.0.gemfile
+  - gemfiles/7.1.gemfile
+
+script: bundle exec rspec
 
 matrix:
   exclude:
-    - rvm: rbx
-      gemfile: gemfiles/4.0.gemfile
-    - rvm: 2.4.0
+    # Rails 3.2
+    - rvm: jruby-19mode
+      gemfile: gemfiles/3.2.gemfile
+    - rvm: 2.5
+      gemfile: gemfiles/3.2.gemfile
+    - rvm: 2.7
+      gemfile: gemfiles/3.2.gemfile
+    - rvm: 3.0
       gemfile: gemfiles/3.2.gemfile
-    - rvm: 2.4.0
+    - rvm: 3.1
+      gemfile: gemfiles/3.2.gemfile
+    - rvm: 3.2
+      gemfile: gemfiles/3.2.gemfile
+    - rvm: 3.3
+      gemfile: gemfiles/3.2.gemfile
+
+    # Rails 4.0
+    - rvm: jruby-19mode
+      gemfile: gemfiles/4.0.gemfile
+    - rvm: 2.7
+      gemfile: gemfiles/4.0.gemfile
+    - rvm: 3.0
+      gemfile: gemfiles/4.0.gemfile
+    - rvm: 3.1
+      gemfile: gemfiles/4.0.gemfile
+    - rvm: 3.2
+      gemfile: gemfiles/4.0.gemfile
+    - rvm: 3.3
+      gemfile: gemfiles/4.0.gemfile
+
+    # Rails 4.1
+    - rvm: jruby-19mode
+      gemfile: gemfiles/4.1.gemfile
+    - rvm: 2.7
       gemfile: gemfiles/4.1.gemfile
-    - rvm: 2.4.0
+    - rvm: 3.0
+      gemfile: gemfiles/4.1.gemfile
+    - rvm: 3.1
+      gemfile: gemfiles/4.1.gemfile
+    - rvm: 3.2
+      gemfile: gemfiles/4.1.gemfile
+    - rvm: 3.3
+      gemfile: gemfiles/4.1.gemfile
+
+    # Rails 4.2
+    - rvm: jruby-19mode
+      gemfile: gemfiles/4.2.gemfile
+    - rvm: 2.7
+      gemfile: gemfiles/4.2.gemfile
+    - rvm: 3.0
+      gemfile: gemfiles/4.2.gemfile
+    - rvm: 3.1
+      gemfile: gemfiles/4.2.gemfile
+    - rvm: 3.2
       gemfile: gemfiles/4.2.gemfile
-  include:
-    - rvm: 2.2.6
+    - rvm: 3.3
+      gemfile: gemfiles/4.2.gemfile
+
+    # Rails 5.0
+    - rvm: jruby-19mode
       gemfile: gemfiles/5.0.gemfile
-    - rvm: 2.3.3
+    - rvm: 2.1
       gemfile: gemfiles/5.0.gemfile
-    - rvm: 2.4.0
+    - rvm: 3.0
       gemfile: gemfiles/5.0.gemfile
+
+    # Rails 5.1
+    - rvm: jruby-19mode
+      gemfile: gemfiles/5.1.gemfile
+    - rvm: 2.1
+      gemfile: gemfiles/5.1.gemfile
+    - rvm: 3.0
+      gemfile: gemfiles/5.1.gemfile
+
+    # Rails 5.2
+    - rvm: jruby-19mode
+      gemfile: gemfiles/5.2.gemfile
+    - rvm: 2.1
+      gemfile: gemfiles/5.2.gemfile
+    - rvm: 3.0
+      gemfile: gemfiles/5.2.gemfile
+
+    # Rails 6.0
+    - rvm: jruby-19mode
+      gemfile: gemfiles/6.0.gemfile
+    - rvm: 2.1
+      gemfile: gemfiles/6.0.gemfile
+    - rvm: 2.3
+      gemfile: gemfiles/6.0.gemfile
+    - rvm: 2.4
+      gemfile: gemfiles/6.0.gemfile
+    - rvm: 2.2
+      gemfile: gemfiles/6.0.gemfile
+    - rvm: 3.0
+      gemfile: gemfiles/6.0.gemfile
+
+    # Rails 6.1
+    - rvm: jruby-19mode
+      gemfile: gemfiles/6.1.gemfile
+    - rvm: 2.1
+      gemfile: gemfiles/6.1.gemfile
+    - rvm: 2.2
+      gemfile: gemfiles/6.1.gemfile
+    - rvm: 2.3
+      gemfile: gemfiles/6.1.gemfile
+    - rvm: 2.4
+      gemfile: gemfiles/6.1.gemfile
+    - rvm: 3.0
```

Comment:
I thought that 3.0 was the [recommended](https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html) version of ruby for rails 6.1? And compatible with 7 & 7.1 too. Unless I'm reading this list wrong. But it looks like 3.0 is excluded for every version of Rails - seems like it should be compatible with a couple versions at least.

Reply:
thanks for catching that. i had some issues with the build not being able to install 3.0 early on. I overlooked going back to revisit this. working on it now

Reply:
fixed it!